### PR TITLE
Allow aliases to be skipped

### PR DESCRIPTION
**Pull request by Edge00**
Refer to https://github.com/Richienb/node-polyfill-webpack-plugin/issues/5

Fixes: #8 #7 

-----
[View rendered readme.md](https://github.com/Edge00/node-polyfill-webpack-plugin/blob/master/readme.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,12 +2,25 @@
...
 	apply(compiler) {
-		compiler.options.plugins.push(new ProvidePlugin({
+		const provides = {
 			Buffer: ["buffer", "Buffer"],
 			console: "console-browserify",
 			process: "process/browser"
-		}))
+		}
+
+		for (const it of this.exclude) {
```
Comment (Richienb) [inline index.js:17]: We can use `filter-obj` here to simplify the implementation

File: index.js [hunk 2]
```diff
@@ -2,12 +2,25 @@
 const { ProvidePlugin } = require("webpack")
 
 module.exports = class NodePolyfillPlugin {
+	constructor(options = {}) {
+		this.options = options
```
Comment (Richienb) [inline index.js:6]: We can use object spread syntax to simplify this

**Review by Richienb — CHANGES_REQUESTED** [review]
Looks good - just some minor nitpicks

File: index.js (hunk 2 above)
Comment (Edge00) [reply index.js:6]: Like this ?
```js
constructor(options = {}) {
  this.options = { exclude: [], ...options}
}
```

Comment (Richienb) [reply index.js:6]: 👍🏻

File: readme.md [hunk 3]
```diff
@@ -26,3 +26,22 @@ module.exports = {
 	]
 }
 ```
+
+## Options
+
+### exclude
+
+Specify aliases to skip adding.
```
Comment (Richienb) [inline readme.md:34]: This should also be included in the typings file

Comment (Edge00) [reply readme.md:34]: Added already!

Comment (Richienb) [reply readme.md:34]: I meant the docstring.

Comment (Edge00) [reply readme.md:34]: Sure. got it.

Comment (Edge00) [reply readme.md:34]:
```ts
import { Compiler } from "webpack"

declare interface NodePolyfillPluginOptions {
	/**
	 * Specify aliases to skip adding.
	 */
	exclude?: string[]
}

declare class NodePolyfillPlugin {
	constructor(options: NodePolyfillPluginOptions)

	apply(compiler: InstanceType<typeof Compiler>): void
}
```

**Comment by Edge00** [comment]
Is there any thing left to do?

**Comment by Richienb** [comment]
There's a couple meta tweaks I gotta do but regardless, I'll get it merged in a day or so.

**Comment by constgen** [comment]
May it will be more useful just to allow Boolean options
```js
new NodePolyfillPluginOptions ({
  Buffer: true,
  console: true,
  process: true
})
```

**Comment by Edge00** [comment]
> May it will be more useful just to allow Boolean options
> 
> ```js
> new NodePolyfillPluginOptions ({
>   Buffer: true,
>   console: true,
>   process: true
> })
> ```

@constgen If this, users must understand what polyfills this plugin provided, and what polyfills their project or project dependency need.

**Pull request merged**